### PR TITLE
feat(benchmark): catch preflight runtime blockers (#5300)

### LINKED ISSUE #5300: benchmark: preflight must catch missing arm deps and unresolvable map_file (would have saved 4 GPU jobs on 2026-07-11)
## Execution packet (fully pre-decided)

**Goal.** Camera-ready **preflight mode must catch what only full runs catch today**: missing per-arm policy dependencies and unresolvable scenario `map_file` references.

**Why now.** On 2026-07-11 preflight job 13370 PASSED while the subsequent runs failed in seconds: 13369/13371 on `stable_baselines3 not installed` (PPO arm deps never imported during preflight) and 13372/13373 on `map_file ... could not be resolved` (maps never resolved during preflight). Four GPU allocations were burned on failures a CPU preflight should have caught.

**Decision (made, do not revisit):** extend preflight (`CAMERA_READY_BENCHMARK_MODE=preflight`) with two checks, both CPU-only and fail-fast with actionable messages:
1. **Arm-dependency probe** — for each enabled planner arm, import the modules its policy path requires (e.g. `stable_baselines3` for ppo/checkpoint arms) in the SAME interpreter/venv the run will use; report the exact `uv sync --all-extras` remediation on failure.
2. **Map resolvability sweep** — for every scenario in the prepared (post-normalization) scenario list, call the existing `resolve_map_definition` with the same `scenario_path` semantics the run uses; fail listing every unresolvable `(scenario, map_file)`.

**Verified anchors** (origin/main):
- preflight entry: `scripts/tools/run_camera_ready_benchmark.py` mode handling; existing checkpoint preflight `robot_sf/benchmark/campaign_checkpoint_preflight.py` (`check_campaign_arm_checkpoints_preflight`, log line at campaign start) — follow its structure for the two new checks
- map resolution: `robot_sf/training/scenario_loader.py::resolve_map_definition` (:1530) and the campaign-side normalization in `robot_sf/benchmark/camera_ready/_config.py::_load_campaign_scenarios` (:534)
- scenario handoff (post-#5270): parent-prepared `scoped_scenarios` are the list to sweep — do NOT re-load the matrix

**Acceptance.** A config whose arm needs a missing import, or whose scenario names a bogus map_file, fails preflight with a named remediation; a valid config still passes. Unit tests for both checks (mock the import for the dep probe).

**Out of scope.** No run-mode behavior changes; no metric semantics (freeze ~07-18); no Slurm wrapper edits.

**Verify:** `uv run pytest tests/benchmark/ -k "preflight" -v`

### PR DESCRIPTION
**Pull request by ll7**
## Reviewer-critical summary

- **What changed:** camera-ready preflight now imports each enabled learned-policy arm's required runtime module and resolves every post-normalization scenario `map_file` before artifacts or benchmark execution begin.
- **Why now:** #5300 recorded four GPU allocations lost to missing `stable_baselines3` and unresolvable maps that a CPU preflight had not exercised.
- **Claim boundary:** this is CPU-only readiness validation; it does not run a benchmark or establish any benchmark, metric, model-provenance, paper, or dissertation result.
- **Required validation:** `uv run pytest tests/benchmark/ -k 'preflight' -v` (282 passed), focused dependency/map and checkpoint-preflight regressions (26 passed), and the final committed-HEAD readiness gate.
- **Remaining blocker:** none for #5300's code contract. A full benchmark campaign remains intentionally out of scope.

## Contract delta

- New fail-closed blockers: an enabled PPO/SAC/guarded/hybrid learned arm missing `stable_baselines3`, or enabled Torch-backed arm missing `torch`, now fails with the named arm, module, and `uv sync --all-extras` remedy.
- New fail-closed blocker: every prepared scenario map must resolve through `resolve_map_definition`; preflight lists all `(scenario, map_file)` failures together.
- Compatibility: valid configs are unchanged. Existing route-clearance fixture SVGs now satisfy the same map-validity prerequisite, and the former warning-only malformed-map expectation correctly fails before route-clearance analysis.

Closes #5300

Issue: https://github.com/ll7/robot_sf_ll7/issues/5300

## Scope and validation

- Full issue implementation: adds the nameable capability **per-arm runtime dependency and prepared-map preflight** to `CAMERA_READY_BENCHMARK_MODE=preflight`'s camera-ready path.
- `uv run pytest tests/benchmark/test_campaign_runtime_preflight.py tests/benchmark/test_campaign_checkpoint_preflight.py tests/benchmark/test_camera_ready_checkpoint_submit_preflight.py -v` — 26 passed.
- `uv run pytest tests/benchmark/ -k 'preflight' -v` — 282 passed, 4,371 deselected.
- `uv run ruff check robot_sf/benchmark/campaign_runtime_preflight.py robot_sf/benchmark/camera_ready/_preflight.py tests/benchmark/test_campaign_runtime_preflight.py tests/benchmark/test_camera_ready_campaign.py tests/benchmark/test_route_clearance_fail_closed.py` — passed.
- `PR_READY_MODE=final BASE_REF=origin/main scripts/dev/pr_ready_check.sh` — passed at committed HEAD (freshness stamp below).

Out of scope: no full benchmark campaign run, no Slurm/GPU submission, and no paper/dissertation claim edits.

<details>
<summary>Governance, artifacts, and propagation appendix</summary>

### Implementation assumption

The reversible mapping probes the policy modules selected by camera-ready arms: Stable-Baselines3 for PPO/SAC/guarded and hybrid route-conditioned policies, and PyTorch for DRL-VO, distributional-RL, and diffusion policies. ORCA retains its existing dedicated `rvo2` preflight. This directly covers the recorded missing-SB3 failure without constructing policies or loading checkpoints.

### Artifacts and state propagation

No generated artifacts are committed; `output/` contains only the readiness freshness stamp and remains ignored. No issue-state comment was added because this task explicitly does not authorize issue or PR comments; this PR body is the canonical reviewable propagation surface.

### Fragmentation, evidence, and friction log

No merged #5300 guardrail/checker/packet-refresh PRs were found in the prior 24 hours, so this is a progression slice with the new capability named above. Downstream research propagation is not applicable: support validation only, with no empirical claim. `friction_issues: []` — the one shared-venv freshness guard provided a documented, successful local-venv setup path, and the route-clearance fixture incompatibility was directly corrected within this issue's test scope.

### Explicit exclusions

- No compute submission, merge, release, or deletion.
- No transient queue-routing state in tracked files.
- No benchmark metric or claim-semantics change.

</details>

### CODE REVIEW

**Comment by ll7** [comment]
## Gate findings — 2026-07-11

Reviewed head `154c64417e9f9cd27210d375cbb7e7db116626ce` against #5300. Intended-design status: **blocked pending one in-scope fail-closed repair**. The CodeRabbit rate-limit/skip notice is advisory; this is the review of record.

Open review-thread ledger: **3 open review threads**.

1. `PRRT_kwDOLRSZdc6QGRq1` (HIGH, `campaign_runtime_preflight.py:182`) — resolver/parser failures outside `OSError` and `ValueError` escape instead of being collected with every invalid prepared map. **Verdict: fixed in phase 2** by aggregating resolver exceptions into the structured fail-closed error; a regression test will cover the non-`ValueError` path.
2. `PRRT_kwDOLRSZdc6QGRq2` (MEDIUM, `campaign_runtime_preflight.py:189`) — the diagnostic formatter retains an unnecessary `None` branch. **Verdict: fixed in phase 2** as a consequence of storing an exception for every map-resolution failure.
3. `PRRT_kwDOLRSZdc6QGRq4` (MEDIUM, `campaign_runtime_preflight.py:166`) — use a more explicit resolver assignment instead of the nested default function. **Verdict: waived**; the current typed nested function correctly preserves the required `scenario_path` binding, and changing its spelling would not improve runtime correctness or the fail-closed contract.

No out-of-scope follow-up is needed. I will re-check the reviewed head, all three threads, focused tests, and CI after the phase-2 repair before deciding merge readiness.

**Comment by ll7** [comment]
## Gate phase-2 delta

Head advanced to `4d33fd4e18863a0c9212258665cd9f01ae7c6f4f`.

- `PRRT_kwDOLRSZdc6QGRq1` — **fixed**: all map resolver/parser exceptions are now aggregated in the structured fail-closed error, with a non-`ValueError` regression case.
- `PRRT_kwDOLRSZdc6QGRq2` — **fixed**: every failure now has an exception, so diagnostics always include its type and message.
- `PRRT_kwDOLRSZdc6QGRq4` — **waived**: the nested typed resolver preserves the required `scenario_path` binding; no correctness change warranted.

**0 open review threads.** Focused tests and `tests/benchmark/ -k 'preflight'` passed locally; final local and remote CI gates are still running for this head.
